pkg/chunked/compression_linux.go

```diff
@@ -55,17 +55,20 @@ type topMetadata struct {
 }
 
 // readEstargzChunkedManifest reads the estargz manifest from the seekable stream blobStream.
-func readEstargzChunkedManifest(blobStream ImageSourceSeekable, blobSize int64, tocDigest digest.Digest) (topMetadata, error) {
+// On error, the second return value is true if a fallback to an alternative (either initializeConvertFromRawDiffer, or a non-partial pull)
+// is permissible.
+func readEstargzChunkedManifest(blobStream ImageSourceSeekable, blobSize int64, tocDigest digest.Digest) (topMetadata, bool, error) {
 	// information on the format here https://github.com/containerd/stargz-snapshotter/blob/main/docs/stargz-estargz.md
 	footerSize := int64(51)
 	if blobSize <= footerSize {
-		return topMetadata{}, errors.New("blob too small")
+		return topMetadata{}, false, errors.New("blob too small")
 	}
 
 	footer := make([]byte, footerSize)
 	streamsOrErrors, err := getBlobAt(blobStream, ImageSourceChunk{Offset: uint64(blobSize - footerSize), Length: uint64(footerSize)})
 	if err != nil {
-		return topMetadata{}, err
+		var badRequestErr ErrBadRequest // If the source does not support partial reads, we might want to fall back.
+		return topMetadata{}, errors.As(err, &badRequestErr), err
```

Comment:
It seems like it might be slightly cleaner to create a new error type e.g. `ErrShouldFallback` that wraps these and the caller tests for that error type?